core/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregator.java

```diff
@@ -122,7 +122,13 @@ public void collect(int docId, long bucket) throws IOException {
                     // In the QueryPhase we don't need this protection, because it is build into the IndexSearcher,
                     // but here we create collectors ourselves and we need prevent OOM because of crazy an offset and size.
                     topN = Math.min(topN, subSearchContext.searcher().getIndexReader().maxDoc());
-                    TopDocsCollector<?> topLevelCollector = sort != null ? TopFieldCollector.create(sort.sort, topN, true, subSearchContext.trackScores(), subSearchContext.trackScores()) : TopScoreDocCollector.create(topN);
+                    TopDocsCollector<?> topLevelCollector;
```

Comment:
Thanks! I think it is pretty rare a ternary that spans multiple lines is more readable than a multi-line if.